### PR TITLE
YD-173 Edit links

### PR DESCRIPTION
**Pull request by aristotelos**

### CODE REVIEW

File: appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java [hunk 1]
```diff
@@ -155,6 +156,7 @@ public GoalResource toResource(GoalDTO goal)
 			GoalResource goalResource = instantiateResource(goal);
 			ControllerLinkBuilder selfLinkBuilder = getGoalLinkBuilder(userID, goal);
 			addSelfLink(selfLinkBuilder, goalResource);
+			addRemoveLink(selfLinkBuilder, goalResource);
```
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/goals/rest/GoalController.java]: Rename into ``addEditLink``, as that is what's added.

File: appservice/src/main/java/nu/yona/server/messaging/rest/MessageController.java [hunk 2]
```diff
@@ -189,6 +194,11 @@ private void addSelfLink(ControllerLinkBuilder selfLinkBuilder, MessageResource
 			messageResource.add(selfLinkBuilder.withSelfRel());
 		}
 
+		private void addDeleteLink(ControllerLinkBuilder selfLinkBuilder, MessageResource messageResource)
```
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/messaging/rest/MessageController.java]: Rename into ``addEditLink``, as that is what's added.

File: appservice/src/main/java/nu/yona/server/subscriptions/rest/BuddyController.java [hunk 3]
```diff
@@ -202,5 +204,10 @@ private void addSelfLink(ControllerLinkBuilder selfLinkBuilder, BuddyResource bu
 		{
 			buddyResource.add(selfLinkBuilder.withSelfRel());
 		}
+
+		private void addRemoveLink(ControllerLinkBuilder selfLinkBuilder, BuddyResource buddyResource)
```
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/subscriptions/rest/BuddyController.java]: Rename into ``addEditLink``, as that is what's added.

File: core/src/main/java/nu/yona/server/messaging/service/DiscloseResponseMessageDTO.java [hunk 4]
```diff
@@ -53,6 +53,12 @@ public String getNickname()
 		return nickname;
 	}
 
+	@Override
+	public boolean canBeDeleted()
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/messaging/service/DiscloseResponseMessageDTO.java]: We should implement a test to verify deletion of such messages. In the original code they could not be deleted, but that wasn't noticed.

**Comment by Bert-R** [comment]
Looks good. A few minor comments added.

**Comment by aristotelos** [comment]
Rework done

**Pull request merged**